Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs

```diff
@@ -66,4 +68,40 @@ public ApiListResponse<ApiLevelCategoryResponse> GetCategories(RequestContext co
         
         return ApiGameLevelResponse.FromOld(level);
     }
+    
+    [ApiV3Endpoint("levels/id/{id}", HttpMethods.Patch)]
+    [DocSummary("Edits a level by the level's numerical ID")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.LevelMissingErrorWhen)]
+    [DocError(typeof(ApiAuthenticationError), ApiAuthenticationError.NoPermissionsForObjectWhen)]
+    public ApiResponse<ApiGameLevelResponse> EditLevelById(RequestContext context, GameDatabaseContext database, GameUser user,
+        [DocSummary("The ID of the level")] int id, ApiEditLevelRequest body)
+    {
+        GameLevel? level = database.GetLevelById(id);
+        if (level == null) return ApiNotFoundError.LevelMissingError;
+
+        if (level.Publisher?.UserId != user.UserId) 
+            return ApiAuthenticationError.NoPermissionsForObject;
+
+        database.UpdateLevel(body, level);
+
+        return ApiGameLevelResponse.FromOld(level);
+    }
+
+    [ApiV3Endpoint("levels/id/{id}", HttpMethods.Delete)]
+    [DocSummary("Deletes a level by the level's numerical ID")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.LevelMissingErrorWhen)]
+    [DocError(typeof(ApiAuthenticationError), ApiAuthenticationError.NoPermissionsForObjectWhen)]
+    public ApiOkResponse DeleteLevelById(RequestContext context, GameDatabaseContext database, GameUser user,
+        [DocSummary("The ID of the level")] int id)
+    {
+        GameLevel? level = database.GetLevelById(id);
+        if (level == null) return ApiNotFoundError.LevelMissingError;
+
+        if (level.Publisher?.UserId != user.UserId) 
+            return ApiAuthenticationError.NoPermissionsForObject;
+
+        database.DeleteLevel(level);
+
+        return new ApiOkResponse();
+    }
```

Comment:
should this be merged with the admin "delete level" request, with the check on line 100 being ignored if the user is an admin?

Reply:
Probably, but I would prefer to keep admin land and user land entirely separate.

Currently it's a check we handle at the service level rather than the actual endpoint code, so if we checked if the user was an admin for each endpoint then we'd be spamming that check everywhere.

It's just easier to manage in my head.